sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -382,6 +429,11 @@ def _setup_failover_endpoints(self):
                     )
         self._replica_clients = new_clients
         self._next_update_time = time.time() + MINIMAL_CLIENT_REFRESH_INTERVAL
+        if updated_endpoints and self._load_balancing_enabled:
+            # Reshuffle only if a new client was added and _load_balancing_enabled is enabled
+            # This allways needs to be shuffled, but if load balancing is enabled than the primary endpoint need to be
+            #  shuffled too.
+            random.shuffle(self._replica_clients)
```

Comment:
If load balancing is  not enabled, we will never shuffle anything according to this code.

Reply:
@amerjusupovic Is it correct that we always shuffle the discovered clients, not just for load balancing? Trying to verify as Spring doesn't shuffle at all, or does the released version of python.

Reply:
That's a good point, we've always had a shuffle for the discovered replicas [here](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/90835cf5efbfb26d6de6068e4c31c3050947b78a/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs#L238-L240). We shuffle and then sort by priority/weight (which we don't use yet but should respect, all hosts have the same values for now).

Reply:
@amerjusupovic do we have plans to add priority/weight support?